Mapping/LiSim/sim.py

```diff
@@ -37,25 +37,26 @@ def save_carte(self):
     def give_carte(self):
         return self.carte
 
-    def noise(self, clean_data):
+    def _noise(self, clean_data, dev, esp=0):
         '''Function that adds noise from a normal distribution to the data provided to mimic real world'''
         data = np.array(clean_data)
-        noise = np.random.normal(self.mu,self.std,[int(360/self.angle_step),2])
+        noise = np.random.normal(esp,dev,clean_data.shape)
         return noise+clean_data
 
-    def simulate(self, show=False):
+    def simulate(self, show=False, uPos=2, uDist=2.5, uTheta=1.25):
         '''Simulates data from one complete lidar rotation'''
         data=[]
         if show == True:
             print(self.carte)
         for i in range(int(360/self.angle_step)):
             current_ray = 1
+            theta = i
             while True:
                 x_current = int(m.cos(m.radians(i))*current_ray)
                 y_current = int(m.sin(m.radians(i))*current_ray)
 
                 if self.carte[x_current+self.pos[0]][y_current+self.pos[1]] == 1:
-                    data.append((current_ray , i))
+                    data.append((self._noise(current_ray, uDist) , self._noise(theta, uTheta) , self._noise(self.pos, uPos)))
```

Comment:
This raises an excpetion, noise takes in an array and you're giving it an integer

Reply:
Yeah should be a one value array, will fix that when I can grab my computer, alternatively one can do it on the final array of arrays column by column

Reply:
Great, commit your changes, submit the pull request and I'll approve it